src/structured_data.rs

```diff
@@ -57,6 +57,19 @@ impl StructuredData {
                    previous_owner_signatures: previous_owner_signatures
                  }
     }
+
+    /// This is a static function required for computing a name give the tag-type and identifier to
+    /// be use by GETs
+    pub fn compute_name(type_tag: u64, identifier: &NameType) -> NameType {
```

Comment:
What is the use-case for having compute_name as a public function, when we already have pub fn name?